packages/web-components/fast-components/src/tabs/tabs.ts

```diff
@@ -0,0 +1,301 @@
+    private tabIds: Array<string | null>;
+    private tabpanelIds: Array<string | null>;
+
+    private change = (): void => {
+        this.$emit("change", this.activetab);
+    };
+
+    private getActiveIndex(): number {
+        const id: string = this.activeid;
+        if (id !== undefined) {
+            return this.tabIds.indexOf(this.activeid) === -1
+                ? 0
+                : this.tabIds.indexOf(this.activeid);
+        } else {
+            return 0;
+        }
+    }
+
+    private setTabs = (): void => {
+        this.tabIds = this.getTabIds();
+        this.tabpanelIds = this.getTabPanelIds();
+        this.activeTabIndex = this.getActiveIndex();
+        this.tabs.forEach((tab: HTMLElement, index: number) => {
+            if (tab.slot === "tab") {
+                const tabID: string | null = this.tabIds[index];
+                const tabpanelId: string | null = this.tabpanelIds[index];
+                tab.setAttribute(
+                    "id",
+                    tabID === undefined || tabID === null ? `tab-${index + 1}` : tabID
+                );
+                tab.setAttribute(
+                    "aria-selected",
+                    this.activeTabIndex === index ? "true" : "false"
+                );
+                tab.setAttribute(
+                    "aria-controls",
+                    tabpanelId === undefined || tabpanelId === null
+                        ? `panel-${index + 1}`
+                        : tabpanelId
+                );
+                tab.setAttribute(
+                    "style",
+                    this.isHorizontal()
+                        ? `grid-column: ${index + 1};`
+                        : `grid-row: ${index + 1};`
+                );
+                tab.addEventListener("click", this.handleTabClick);
+                tab.addEventListener("keydown", this.handleTabKeyDown);
+                tab.setAttribute("tabindex", this.activeTabIndex === index ? "0" : "-1");
+                if (this.activeTabIndex === index) {
+                    this.activetab = tab;
+                }
+                !this.isHorizontal()
+                    ? tab.classList.add("vertical")
+                    : tab.classList.remove("vertical");
+            }
+        });
+    };
+
+    private setTabPanels = (): void => {
+        this.tabIds = this.getTabIds();
+        this.tabpanelIds = this.getTabPanelIds();
+        this.tabpanels.forEach((tabpanel: HTMLElement, index: number) => {
+            const tabID: string | null = this.tabIds[index];
+            const tabpanelId: string | null = this.tabpanelIds[index];
+            tabpanel.setAttribute(
+                "id",
+                tabpanelId === undefined || tabpanelId === null
+                    ? `panel-${index + 1}`
+                    : tabpanelId
+            );
+            tabpanel.setAttribute(
+                "aria-labeledby",
+                tabID === undefined || tabID === null ? `tab-${index + 1}` : tabID
```

Comment:
minor nit here, but you could do this with only one comparison:

`typeof tabID === "string"`